internal/awsservice/cloudwatchlogs.go

```diff
@@ -107,26 +52,45 @@ func DeleteLogGroup(logGroupName string) {
 	}
 }
 
-// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
-// It should query the given log stream for log events, and then confirm that the log lines that are
-// returned match the expected log lines. This also sanitizes the log lines from both the output and
-// the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
-func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
-	log.Printf("Checking %s/%s since %s for %d expected logs", logGroup, logStream, since.UTC().Format(time.RFC3339), len(logLines))
+// ValidateLogs queries a given LogGroup/LogStream combination given the start and end times, and executes an
+// arbitrary validator function on the found logs.
+func ValidateLogs(logGroup, logStream string, since, until *time.Time, validator func(logs []string) bool) (bool, error) {
+	log.Printf("Checking %s/%s\n", logGroup, logStream)
+
+	foundLogs, err := getLogsSince(logGroup, logStream, since, until)
+	if err != nil {
+		return false, err
+	}
+
+	return validator(foundLogs), nil
+}
 
-	sinceMs := since.UnixNano() / 1e6 // convert to millisecond timestamp
+// getLogsSince makes GetLogEvents API calls, paginates through the results for the given time frame, and returns
+// the raw log strings
+func getLogsSince(logGroup, logStream string, since, until *time.Time) ([]string, error) {
+	foundLogs := make([]string, 0)
 
 	// https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html
 	// GetLogEvents can return an empty result while still having more log events on a subsequent page,
 	// so rather than expecting all the events to show up in one GetLogEvents API call, we need to paginate.
 	params := &cloudwatchlogs.GetLogEventsInput{
 		LogGroupName:  aws.String(logGroup),
 		LogStreamName: aws.String(logStream),
-		StartTime:     aws.Int64(sinceMs),
 		StartFromHead: aws.Bool(true), // read from the beginning
 	}
 
-	foundLogs := make([]string, 0)
+	var sinceMs int64
+	if since != nil {
+		sinceMs = since.UnixNano() / 1e6 // convert to millisecond timestamp
```

Comment:
I see this is how it was even before your changes, but wonder if there is a specific reason to do it this way instead of just using `since.UnixMilli()`.